ffi/framebuffer_linux.lua

```diff
@@ -36,9 +36,12 @@ function framebuffer:init()
         self.fb_size = vinfo.xres_virtual * vinfo.yres_virtual * vinfo.bits_per_pixel / 8
     -- Newer eink framebuffer (Kindle Touch, Paperwhite, Kobo)
     elseif ffi.string(finfo.id, 11) == "mxc_epdc_fb" then
-        -- it seems that finfo.smem_len is unreliable on kobo
         -- Figure out the size of the screen in bytes
         self.fb_size = vinfo.xres_virtual * vinfo.yres_virtual * vinfo.bits_per_pixel / 8
+        -- Kobo hasn't updated smem_len when the depth was changed from 16 to 32
+        if self.fb_size > finfo.smem_len then
```

Comment:
I suppose you mean the resulting behavior is equivalent in a 32-bit scenario, but based on the comment you removed (why?) it looks like my first thought was bad regardless. Unfortunately I have no idea what exactly was meant by `smem_len` being unreliable.

In what way does `vinfo.xres_virtual * vinfo.yres_virtual * vinfo.bits_per_pixel / 8` differ from `finfo.smem_len`? Is it half the size as you would expect when `vinfo.bits_per_pixel` is 16 or is it different in another way? Regardless of the result, does `vinfo.xres` (as opposed to `_virtual`) work because you mentioned `xres` is `1404` and `xres_virtual` is `1408`?

Basically just from looking at this I'm not sure if `self.fb_size > finfo.smem_len` is sufficiently generic. If it's not, I'm inclined to think the test should be something more specific like `vinfo.bits_per_pixel == 32`.